### PR TITLE
Telemetry for Query Execution Plans

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR adds telemetry around the following areas for execution plans:
- execution plan properties window,
- execution plan zoom functionality,
- execution plan top operations tab
- execution plan open query option
- execution plan node searching

### CODE REVIEW

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts [hunk 1]
```diff
@@ -273,11 +275,18 @@ export class OpenQueryAction extends Action {
...
+	constructor(private triggeredFromContextMenu: boolean,
+		@IAdsTelemetryService private readonly telemetryService: IAdsTelemetryService
+	) {
 		super(OpenQueryAction.ID, OpenQueryAction.LABEL, openQueryIconClassNames);
 	}
 
 	public override async run(context: ExecutionPlanView): Promise<void> {
+		this.telemetryService
+			.createActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.OpenQuery)
+			.withAdditionalProperties({ openedFromContextMenu: this.triggeredFromContextMenu, openedFromActionBar: !this.triggeredFromContextMenu })
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:287]: having one property: source will make report building easier.

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts [hunk 2]
```diff
@@ -273,11 +275,20 @@ export class OpenQueryAction extends Action {
...
 
-	constructor() {
+	constructor(private triggeredFromContextMenu: boolean,
+		@IAdsTelemetryService private readonly telemetryService: IAdsTelemetryService
+	) {
 		super(OpenQueryAction.ID, OpenQueryAction.LABEL, openQueryIconClassNames);
 	}
 
 	public override async run(context: ExecutionPlanView): Promise<void> {
+		let source = this.triggeredFromContextMenu ? 'Context menu' : 'Action bar';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:285]: Have the constructor take in a string that's the source instead

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts (hunk 2 above)
Comment (alanrenmsft) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:285]: please define a type for the source.

`export type ExecutionPlanActionSource = 'ContextMenu'|'ActionBar';`

Comment (lewis-sanchez) [reply src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:285]: Do I export the new type from here or a different file?

Comment (alanrenmsft) [reply src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:285]: in this file should be fine. not a huge deal.

File: src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts [hunk 3]
```diff
@@ -269,15 +271,24 @@ export class ExecutionPlanView implements ISashLayoutProvider {
 	}
 }
 
+export type ExecutionPlanActionSource = 'ContextMenu' | 'ActionBar';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/executionPlan/browser/executionPlanView.ts:274]: This looks like it doesn't need to be exported

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**